test/parallel/test-timers-clearImmediate.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+var count = 0;
+function next() {
+  const immediate = setImmediate(function() {
+    clearImmediate(immediate);
+    ++count;
+  });
+}
+for (var i = 0; i < 3; ++i)
+  next();
```

Comment:
It wasn't really recursive, just restarting an immediate timer. However, that's not necessary to reproduce the original issue. The issue just has to do with calling `clearImmediate()` from within an immediate callback when there are other items in the immediate queue.